### PR TITLE
PR for #134: Notifications

### PR DESCRIPTION
**Pull request by matthewgallo**
Contributes to #134 

This adds the new Notifications component, it is based off of the designs seen [here](https://pages.github.ibm.com/cdai-design/pal/patterns/notifications/#notifications-view-all-page). This code is only for the Notifications panel, in the design there is also an entire page for Viewing all notifications. This component also makes use of the new `<EmptyState />` component, when there are not any notifications.

#### Changelog

`packages/experimental/src/components/Notifications/index.scss`
`packages/experimental/src/components/Notifications/_notifications.scss`
`packages/experimental/src/components/Notifications/index.js`
`packages/experimental/src/components/Notifications/Notifications.stories.js`
`packages/experimental/src/components/Notifications/Notifications_data.js`
`packages/experimental/src/components/Notifications/Notifications-test.js`
`packages/experimental/src/components/Notifications/Notifications.js`
`packages/experimental/src/components/Notifications/utils.js`
`packages/experimental/src/components/Notifications/Notificaitons.mdx`

**New**

- Added new Notifications component based on designs seen [here](https://pages.github.ibm.com/cdai-design/pal/patterns/notifications)

#### Testing / Reviewing

Run `yarn test` and see that all tests pass for this component.

image

### CODE REVIEW

File: packages/experimental/src/components/Notifications/Notifications.js [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+              Dismiss all
+            </Button>
+          </div>
+          <ToggleSmall
+            className={`${prefix}-notifications-do-not-disturb-toggle`}
+            id={`${prefix}-notifications-do-not-disturn-toggle-component`}
+            labelA="Do not disturb"
+            labelB="Do not disturb"
+            onToggle={(event) => onDisableNotificationChange(event)}
+            aria-label="Do not disturb toggle"
```
Comment (lee-chase) [inline packages/experimental/src/components/Notifications/Notifications.js:207]: Is an aria-label needed when other labels are visible?

File: packages/experimental/src/components/Notifications/Notifications.js [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+              }}>
+              Dismiss all
+            </Button>
+          </div>
+          <ToggleSmall
+            className={`${prefix}-notifications-do-not-disturb-toggle`}
+            id={`${prefix}-notifications-do-not-disturn-toggle-component`}
+            labelA="Do not disturb"
+            labelB="Do not disturb"
+            onToggle={(event) => onDisableNotificationChange(event)}
```
Comment (lee-chase) [inline packages/experimental/src/components/Notifications/Notifications.js:206]: Event should be 'onDoNotDisturbChange' to match other naming.

File: packages/experimental/src/components/Notifications/Notifications.js [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+
+  return (
+    shouldRender && (
+      <div
+        className={[`${prefix}-notifications-panel-container`].join(' ')}
+        style={{ animation: `${open ? 'fadeIn 250ms' : 'fadeOut 250ms'}` }}
+        onAnimationEnd={onAnimationEnd}>
+        <div className={`${prefix}-notifications-header-container`}>
+          <div className={`${prefix}-notifications-header-flex`}>
+            <h6 className={`${prefix}-notifications-header`}>Notifications</h6>
```
Comment (lee-chase) [inline packages/experimental/src/components/Notifications/Notifications.js:190]: A h6?

**Review by lee-chase — CHANGES_REQUESTED** [review]
- * Please do not use Knobs in your story, switch to use Controls instead which is largely automatic if you include Proptypes.
- Animate reveal/close when clicking read more.
- Should this close on click outside?
- Can we show a notification being received, perhaps on a timer?
- * All text (labels, titles, aria-labels, content) should be from properties.
- How will this work in conjunction with the HeaderGlobalAction notification icon? Will we be providing a separate component to display a status there?
- How does do not disturb work?
- Can we use a decorator to show this in context with a UI shell header and notification icon?

* These need addressing, the other items can be questions asked of your designer (Slack and in Issue).

File: packages/experimental/src/components/Notifications/Notifications.js (hunk 3 above)
Comment (matthewgallo) [reply packages/experimental/src/components/Notifications/Notifications.js:190]: I used this just to get the right type size, but I can change it to an `h1` and use the carbon mixin to get the correct type size that way.

File: packages/experimental/src/components/Notifications/Notifications.js (hunk 1 above)
Comment (matthewgallo) [reply packages/experimental/src/components/Notifications/Notifications.js:207]: Not 100% sure I just thought it was safest to add it just in case.

**Comment by matthewgallo** [comment]
Just to address some points of your points @lee-chase:
- I think it is a good idea for this to close when clicking outside
- The Do not disturb functionality should probably be left up to the consumer of this component, that's why I created the `onDoNotDisturbChange` prop that returns the current value of the toggle so they can make whatever api request they need to "quiet" the notifications.
- Yes, I can add a new story that automatically generates new notifications coming in

**Review by lee-chase — APPROVED** [review]
LGTM although you are causing a scroll bar on open.

**Pull request merged**